arcgis-ios-sdk-samples/Maps/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -26,29 +26,90 @@ class DisplayDeviceLocationWithNMEADataSourcesViewController: UIViewController {
+    let supportedProtocolStrings: [String] = {
+        guard let protocols = Bundle.main.object(forInfoDictionaryKey: "UISupportedExternalAccessoryProtocols") as? [String] else {
+            return []
+        }
+        return protocols
+    }()
+    
     // MARK: Instance properties
     
     /// An NMEA location data source, to parse NMEA data.
-    let nmeaLocationDataSource = AGSNMEALocationDataSource(receiverSpatialReference: .wgs84())
+    var nmeaLocationDataSource: AGSNMEALocationDataSource?
+    /// A string to hold the latest satellite info.
+    var satelliteInfoText = "Satellites info will be shown here."
     /// A mock data source to read NMEA sentences from a local file, and generate
     /// mock NMEA data every fixed amount of time.
     let mockNMEADataSource = SimulatedNMEADataSource(nmeaSourceFile: Bundle.main.url(forResource: "Redlands", withExtension: "nmea")!, speed: 1.5)
+    /// A formatter for the accuracy distance string.
+    let distanceFormatter: MeasurementFormatter = {
+        let formatter = MeasurementFormatter()
+        formatter.unitOptions = .naturalScale
+        formatter.numberFormatter.minimumFractionDigits = 1
+        formatter.numberFormatter.maximumFractionDigits = 1
+        return formatter
+    }()
     
     // MARK: Actions
     
-    @IBAction func start() {
+    @IBAction func chooseDataSource(_ sender: UIBarButtonItem) {
+        let alertController = UIAlertController(
+            title: "Choose an NMEA data source.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        // Populate from connected GNSS surveyor devices.
+        EAAccessoryManager.shared().connectedAccessories.forEach { accessory in
+            // The protocol string to establish the EASession.
+            guard let protocolString = accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) }) else {
+                // Skip any device which protocol is not included in the plist.
+                // This typically shouldn't happen, unless the device requires
+                // additional configuration.
+                return
+            }
+            let action = UIAlertAction(title: accessory.name, style: .default) { [self] _ in
+                nmeaLocationDataSource = AGSNMEALocationDataSource(eaAccessory: accessory, protocol: protocolString)
+                nmeaLocationDataSource?.locationChangeHandlerDelegate = self
+                start()
+            }
+            alertController.addAction(action)
+        }
+        // Add mock data source to the options.
+        let mockDataSourceAction = UIAlertAction(title: "Mock Data", style: .default) { [self] _ in
+            nmeaLocationDataSource = AGSNMEALocationDataSource(receiverSpatialReference: .wgs84())
+            nmeaLocationDataSource!.locationChangeHandlerDelegate = self
+            mockNMEADataSource.delegate = self
+            start()
+        }
```

Comment:
Instead of trying to make the mock data appear as an accessory, it might be better to first prompt the user for what _kind_ of data source they want to use: accessory or mock data.